pkg/kademlia/queue.go

```diff
@@ -4,57 +4,108 @@
 package kademlia
 
 import (
+	"container/heap"
 	"math/big"
 
+	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/pb"
 )
 
-// An Item is something we manage in a priority queue.
-type Item struct {
+//We've encapsulated our priorityQueue{} implementation (below) so that you don't
+//have to know how the Golang heap API works (nor our priority XOR logic)
+
+//XorQueue is a priority queue where the priority is key XOR distance
+type XorQueue struct {
+	pq *priorityQueue
+}
+
+//NewXorQueue returns a priorityQueue with priority based on XOR from targetBytes
+func NewXorQueue(nodes []*pb.Node, target dht.NodeID) XorQueue {
```

Comment:
`NewXorQueue(limit int) XorQueue` and use methods to add values.

Reply:
agreed